packages/integration-objects/srv/logging/AuditLogWriter.js

```diff
@@ -0,0 +1,88 @@
+/* Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. */
+const util = require('@sap-cloud-sdk/util');
+const xsenv = require('@sap/xsenv');
+const auditLog = require('../utils/AuditLog');
+const Constants = require('../utils/Constants');
+const { logger } = require('./Logger');
+const uaaData = xsenv.getServices({ xsuaa: { tag: 'xsuaa' } }).xsuaa;
+
+async function writeAuditLog(req, auditParams, currentValues, objectType) {
+  try {
+    const entityType = `${req.event} ${objectType}`;
+    const clientId = uaaData.clientid;
+    const logonUser = req.user ? req.user.id : clientId;
+    const accessedObject = {
+      type: entityType,
+      id: { key: req.data.ID }
+    };
+
+    const configChangeMessage = auditLog.configurationChange(accessedObject);
+    setConfigMessage(req, auditParams, configChangeMessage, currentValues);
+    configChangeMessage.by(logonUser);
+    await auditLog.logPrepare(configChangeMessage);
```

Comment:
I just went over this method, because it was the first one an I thing the `auditLog.logPrepare` is not async. This is the only async method so the whole `writeAuditLog` does not need to be async. Please check all your methods for such mistakes.

Also note that if you use TypeScript you can not make these errors because the compiler tells you that await on a normal method is not possible. 

As I mentioned before could you create an issue for making all the `logging,utils,validations` typescript files. Then you transpile to JS and require as usual in your srv implementation the JS files.

Reply:
Updated the issue and removed the await and async.